### ISSUE TITLE
Retag to latest of BuildConfig not only for DeploymentConfig

### ISSUE THREAD

**Issue by KevinBusse** [opendevstack/ods-jenkins-shared-library#616]
**Is your feature request related to a problem? Please describe.**
For `DeploymentConfig`s and their respective `BuildConfig` I can use `latest` when I describe the image.
The image is build using the git short hash, and retagged as `latest`.

But when I have a `Job` and it's respective `BuildConfig` using `latest` when describing the image it doesn't work.
The image is build using the git short hash, but not retagged as latest.
Neither the Job template is updated.

**Describe the solution you'd like**
Also look for `Job`s, not only for `DeploymentConfig`s when retagging to latest. 

**Describe alternatives you've considered**
Current workarounds are:
* manually retagging the build image to latest
* using the `imageTag` property in the build config to always build directly to latest

**Comment by clemensutschig** [comment]
Dear @KevinBusse  - thank you for the enhancement :)

Just to get it clear - you have a dc and a job - that use the image?

**Comment by KevinBusse** [comment]
No, the image is a job only, no running containers use the same image.

**Comment by clemensutschig** [comment]
hmmm .... I guess here you are (for now) on your own - when it comes to tagging ... (but send a PR up .. ) ;D

**Comment by michaelsauter** [comment]
I think the two alternatives you mention work well.

I see one other option, and that is controlling the Job resource in an OpenShift template, and then updating the image reference in the template with the concrete image tag (equal to the short Git sha) instead of using latest.
